common/src/sim_config.rs

```diff
@@ -28,8 +28,22 @@ pub struct SimConfigRaw {
     pub no_clip_movement_speed: Option<f32>,
     /// Character maximumum movement speed while on the ground in m/s
     pub max_ground_speed: Option<f32>,
+    /// Character artificial speed cap to avoid overloading the server in m/s
+    pub speed_cap: Option<f32>,
+    /// Maximum ground slope (0=horizontal, 1=45 degrees)
+    pub max_ground_slope: Option<f32>,
     /// Character acceleration while on the ground in m/s^2
     pub ground_acceleration: Option<f32>,
+    /// Character acceleration while in the air in m/s^2
+    pub air_acceleration: Option<f32>,
+    /// Acceleration of gravity in m/s^2
+    pub gravity_acceleration: Option<f32>,
+    /// Linear air resistance in (1/e)-lives per second
+    pub air_resistance: Option<f32>,
+    /// How fast the player jumps off the ground in m/s
+    pub jump_speed: Option<f32>,
+    /// How far away the player needs to be from the ground in meters to be considered in the air
+    pub ground_distance_tolerance: Option<f32>,
```

Comment:
Could we `#[derive(Serialize, Deserialize)]` for `CharacterConfig` and just use that here? Might play nicely with `#[serde(default)]`, too.

Reply:
I think that's a good idea, but I won't have time to get to this for a while. I plan to update this PR implementing your other suggestions. I chose not to prioritize this because I don't think it will affect how understandable the PR is to review.

Reply:
I finally have time to work on this again, so I made the suggested change and moved `CharacterConfig` into the `sim_config` module.

Reply:
Original question still stands -- could we have a `pub character: CharacterConfig` field here rather than effectively writing the struct twice?

Reply:
Oh, I had misread your suggestion. I did create `CharacterConfig`, but I didn't create `CharacterConfigRaw`.

I don't think I can make this into `pub character: CharacterConfig` because of the unit-conversion (multiplication by `meters_to_absolute`). There might be a serde feature I'm unaware of, though.

Reply:
Oh, I missed that they were different units. All the workarounds I can immediately think of are similarly annoying to the status quo, unfortunately. No specific need to add a `...Raw` variant for its own sake, though maybe the uniformity would be nice.

Reply:
I went ahead and added a `CharacterConfigRaw` for uniformity, as it should hopefully make custom config files easier to read as well.

Reply:
> All the workarounds I can immediately think of are similarly annoying to the status quo, unfortunately.

I'm not too surprised. If serde had a fancy feature that allowed the unit conversion to be done, that feature would require either backtracking or storing intermediate values somewhere, which likely falls outside its design constraints.

Thinking aloud because it definitely _feels_ like there should be a better way.

One could imagine some fancy newtype-based system to store the meter-based config values originally but then replace those data slots with the absolute-unit-based values (without the code repetition we currently have), but I don't know how to do that in Rust, and I don't know whether it's possible without using custom macros.

If there is something like a `#[derive(Into)]`, that converts corresponding fields, the newtype idea could potentially work with the help of generics, but even then, we would still need to have a version that takes in some kind of context so that we can pass in `meters_to_absolute`. Even if it is possible, it seems overengineered (especially considering the rule of 3, and the fact that `SimConfig` may be altered in-game in the future).

I agree that it's probably not worth pursuing now. Perhaps if/when this repetition becomes unwieldy as additional configuration options are added, it would be worth revisiting.

Reply:
I don't usually take it very seriusly, but the frunk crate's labeled generics might be handy here to at least automate the conversion... but of comical overkill though.

Reply:
Interesting, I haven't heard of the frunk crate before, so I might check it out sometime.